### PR TITLE
Parser: When u flag is set, throw error on invalid unicode escape

### LINKED ISSUE #8: `/\e/u` should throw
`/\e/u` is an invalid regular expression: https://mathiasbynens.be/notes/es6-unicode-regex#impact-syntax

So instead of parsing, I’d expect this to throw:

```
$ regexp-tree -e '/\e/u'
{
  "type": "RegExp",
  "body": {
    "type": "Char",
    "value": "e",
    "kind": "simple",
    "escaped": true
  },
  "flags": [
    "u"
  ]
}
```

### PR DESCRIPTION
**Pull request by Golmote**
This should fix #8

### CODE REVIEW

File: src/parser/regexp.bnf [hunk 1]
```diff
@@ -103,7 +103,14 @@ GROUP_NAME              ([\w$]|\\'u'[0-9a-fA-F]{4}|\\'u{'[0-9a-fA-F]{1,}'}')+
 
 {ESC}'/'                            return 'ESC_CHAR'
 
-{ESC}{CHAR}                         return 'ESC_CHAR'
+{ESC}{CHAR}                         {
+                                        const s = this.getCurrentState();
+                                        if (s === 'u' || s === 'xu' || s === 'u_class') {
+                                            throw new SyntaxError(`invalid unicode escape ${yytext}`);
```
Comment (mathiasbynens) [inline src/parser/regexp.bnf]: Nit: `s/unicode/Unicode/`

Comment (mathiasbynens) [reply src/parser/regexp.bnf]: If this is only about `\p{…}`, how about: “invalid Unicode property escape”?

Comment (Golmote) [reply src/parser/regexp.bnf]: This is not only about `\p{...}`. All redondant escapes (`\e`, `\a`, `\g`...) will throw this error when `u` flag is set.

File: src/parser/__tests__/parser-basic-test.js [hunk 2]
```diff
@@ -1068,4 +1068,10 @@ describe('basic', () => {
     });
   });
 
+  it('throws error on invalid unicode escape', () => {
```
Comment (mathiasbynens) [inline src/parser/__tests__/parser-basic-test.js]: s/unicode/Unicode/

**Comment by DmitrySoshnikov** [comment]
Great! With introducing all `*u` parsing states makes fixing all the Unicode issues in very elegant ways!

**Pull request merged**